modules/grpc/otel/otel-source.cpp

```diff
@@ -259,6 +259,8 @@ otel_sd_new(GlobalConfig *cfg)
   s->super.worker_options.super.stats_source = stats_register_type("opentelemetry");
   s->super.format_stats_key = _format_stats_key;
   s->super.worker_construct = _construct_worker;
+  s->super.worker_options.super.keep_hostname = TRUE;
```

Comment:
Why do we treat otel special in terms of keeping the hostname?

I think this override should be done only for `syslog-ng-otlp()`, but not for the general otel source.